grep-00XX-relicense-volk.md

```diff
@@ -0,0 +1,86 @@
+Especially, long term VOLK contributors expressed their desire to be able to use VOLK in a broader set of projects.
+
+### State of the GREP
+This GREP does currently serve as a platform to collect all the necessary information and ideas to relicense VOLK.
+Everyone is invited to contribute to this GREP with ideas, pull requests, comments etc.
+
+## Copyright / License
+
+This GREP is licensed as CC-BY-ND.
+Copyright 2021 Johannes Demel and any co-author listed above.
+
+
+## Motivation
+
+We recognize the desire of our long term contributors to be able to use VOLK with their contributions in their projects.
+This may explicitly include proprietary projects.
+We want to enable all contributors to be able to use VOLK wherever they want.
+At the same time, we want to make sure that improvements to VOLK itself are easily obtained by everyone, i.e. strike a balance between permissiveness and strict copyleft.
+Previously, several attemps were made to relicense VOLK without success.
+With a formal process and a clear vision laid out, we hope to be able to get everyone on board.
+
+Since VOLK is a library it should choose a fitting license.
+If we see greater adoption of VOLK in more project, we hope to also see more contributions.
+May it be bug fixes, new kernels or even contributions to core features.
+
+## Description
+
+In order to facilitate this relicensing effort, we need to got through several steps.
+
+### Notification
+We consider it to be very important to let every contributor know that the wish to relicense by some contributors exists.
+A first step would be to notify every contributor of this GREP.
+
+### Another license discussion
+Currently, the proposed license is LGPLv3+.
+Though, other licenses, possibly more permissive are available too.
+This includes [3-Clause BSD](https://opensource.org/licenses/BSD-3-Clause), [MIT](https://opensource.org/licenses/MIT) and [Apache-2.0](https://www.apache.org/licenses/LICENSE-2.0).
+Prelimenary discussion on the subject matter revealed that Apache-2.0 is out of favor by some.
+We assume that LGPLv3+ strikes a balance between permissiveness and obligation to provide modified source code.
+
+### All contributors need to sign off for the new license
+
+Before we can adopt a new license, we need approval by every contributor to basically resubmit their contributions under the new license.
+[There is more to this discussion and how this evolved but others are more qualified to add to that discussion]
+
+### The process
+We need to figure out how we collect all the contributor approvals.
+This might be a file within the VOLK repo where everyone adds a signed-off commit to approve the new license.
+Currently, we have around 65 contributors to VOLK.
+
+### The meantime
+It might take time to gain approval by every contributor.
+We want to be able to continue VOLK development while we are in this license transition period.
```

Comment:
Does that entail a CMake flag that disables all non-permissive code, so that people can actually use the parts of VOLK that are already available under non-strong-copyleft terms?

Reply:
If the transition period is, e.g. a month, I'd argue it is more sensible to wait for enough approvals. If it was a year or so, this might justify the additional work with a CMake flag. I'd really like to avoid that though.

Reply:
I think this is going to take a while.

That said, I'm also not sure it's worth trying to split the distribution to have GPL and non-GPL versions. May be better just to wait until you have everything you need and then flip over in one atomic motion.

Reply:
I updated the text here to be more precise.

I have no idea how long things will take. I just hope it won't take too long.

Reply:
I think it'll take a while to get everyone's approval, but it might not take too long to get a few core contributor's approval sooner. That would give us a good core to build upon.

Reply:
For the stragglers, and those who don't want to relicense, I think it's the right way to nuke those kernels or contributions. If people need them, they can contribute new versions thereof.

Reply:
I think it'll take a month to get the vast majority of contributors in place; there'll be a long tail for another 95% & some just never will. I agree @mbr0wn that we need to be willing and able to remove kernels for folks who do not agree after some amount of time -- 2 months maybe -- so as to allow for VOLK development to move forward without splitting releases between GPL and LGPL versions ... which I think will be a PITA.